x/staking/keeper/msg_server.go

```diff
@@ -354,6 +353,28 @@ func (k msgServer) Undelegate(goCtx context.Context, msg *types.MsgUndelegate) (
 		return nil, err
 	}
 
+	validator, found := k.GetValidator(ctx, addr)
+	if !found {
+		return nil, sdkstaking.ErrNoValidatorFound
+	}
+
+	delegation, found := k.GetDelegation(ctx, delegatorAddress, addr)
+	if !found {
+		return nil, status.Errorf(
+			codes.NotFound,
+			"delegation with delegator %s not found for validator %s",
+			msg.DelegatorAddress, msg.ValidatorAddress,
+		)
+	}
+
+	// tokenize share vs exempt delegation check if exempt delegation
+	if delegation.Exempt {
+		maxTokenizeShareAfter := validator.TotalExemptShares.Sub(shares).Mul(k.ExemptionFactor(ctx))
+		if maxTokenizeShareAfter.GT(validator.TotalTokenizedShares) {
+			return nil, types.ErrInsufficientExemptShares
```

Comment:
Shouldn't a delegator always be able to undelegate from a not bonded validator (outside of the validator set)?

Reply:
The tokenized shares need to be redeemed before any exempt delegation to a validator that was out of the active set can undelegation.

Reply:
Now that I understand that the exempt delegations are supposed to be from the validator team, I think it's fine not being able to undelegate when the validator is out of the active set.